### ISSUE TITLE
[Android] Does it works with Android 8,9 notification channels?

### ISSUE THREAD

**Issue by ASerga** [Pushwoosh/pushwoosh-react-native-plugin#49]
I gave up to make it works with Android 8 and 9

In native Android SDK docs there is info that Notification channels work out-of-the-box.
https://www.pushwoosh.com/docs/notification-channels
I sent a push like in docs and tried to find any pushwoosh-related data in "Logcat" but I found nothing! 

In react native docs - nothing about Notification Channel support!
https://www.pushwoosh.com/docs/react-native-plugin

In "pushwoosh-react-native-sample" codebase - nothing about Notification Channel support!
https://github.com/Pushwoosh/pushwoosh-react-native-sample 

---------------
I use "pushwoosh-react-native-plugin==5.9.1" and got Logcat logs like this
`D/Pushwoosh: [RequestManager] registerDevice response success"`
Also it works with Android < 8

------------------

So, questions:

1. Does it work with Notification Channels?
2. And if YES - any example or documentation?

**Comment by ASerga** [comment]
After reproducing with raw RN repository `react-native init PushwooshRN --version react-native@0.57.7` - it works. (could not build Android with RN 0.58)

I will find what Android settings or react-native packages break it, will give feedback and will close the issue.

**Comment by wfhm** [comment]
@ASerga Did you have a chance to figure out what was the breaking step in your build process?

**Comment by ASerga** [comment]
Yes, @wfhm.

Should be installed this NPM packages
- react-native-firebase with FCM
- react-native-intercom

It happens after extending messaging service using this manual.
https://github.com/tinycreative/react-native-intercom#heres-an-example-if-youre-using-react-native-firebase-as-your-existing-fcm-setup

Currently, I just install `react-native-intercom` without Pushes support on Android (the step above is skipped).
If you have any ideas on how to integrate Intercom pushes + don't break Android 8+ pushes - it will be helpful.

**Comment by wfhm** [comment]
@ASerga 

[This service](https://github.com/tinycreative/react-native-intercom#heres-an-example-if-youre-using-react-native-firebase-as-your-existing-fcm-setup) routes push messages between Firebase and Intercom SDKs, and it does not route it to Pushwoosh at all. You should also route Pushwoosh messages to our SDK in it. 

I'd like to suggest creating a custom FirebaseMessagingService extension using the code from Intercom guide and the code from the guide below:

https://www.pushwoosh.com/docs/android-faq#section-using-pushwoosh-sdk-with-other-fcm-services

Basically you should just call `PushwooshFcmHelper.onMessageReceived(this, remoteMessage);` to pass a push to our SDK and handle it whenever you want.

**Comment by ASerga** [comment]
Thanks a lot, @wfhm for a suggestion.
I closed this issue.
